db/files.ts

```diff
@@ -93,7 +93,8 @@ export const createFile = async (
 ) => {
   let validFilename = fileRecord.name.replace(/[^a-z0-9.]/gi, "_").toLowerCase()
   const extension = file.name.split(".").pop()
-  const baseName = validFilename.substring(0, validFilename.lastIndexOf("."))
+  const extensionIndex = validFilename.lastIndexOf(".")
+  const baseName = validFilename.substring(0, (extensionIndex < 0) ? undefined : extensionIndex)
```

Comment:
Some fileRecord.name variables won't have an extension because it is derrived from a filename field in the form. We will have to use file.name to get the extension.